app/src/main/java/mozilla/lockbox/presenter/AutofillRoutePresenter.kt

```diff
@@ -101,51 +121,61 @@ class AutofillRoutePresenter(
 
         val transition = findTransitionId(srcId, destinationId) ?: destinationId
 
-        if (transition == destinationId) {
+        val navOptions = if (transition == destinationId) {
             // Without being able to detect if we're in developer mode,
             // it is too dangerous to RuntimeException.
             val from = activity.resources.getResourceName(srcId)
             val to = activity.resources.getResourceName(destinationId)
+            val graphName = activity.resources.getResourceName(navController.graph.id)
             log.error(
                 "Cannot route from $from to $to. " +
-                    "This is a developer bug, fixable by adding an action to graph_autofill.xml"
+                    "This is a developer bug, fixable by adding an action to $graphName.xml and/or ${javaClass.simpleName}"
             )
+            null
         } else {
-            val clearBackStack = src.getAction(transition)?.navOptions?.shouldLaunchSingleTop() ?: false
-            if (clearBackStack) {
-                while (navController.popBackStack()) {
-                    // NOP
+            // Get the transition action out of the graph, before we manually clear the back
+            // stack, because it causes IllegalArgumentExceptions.
+            src.getAction(transition)?.navOptions?.let { navOptions ->
+                if (navOptions.shouldLaunchSingleTop()) {
+                    while (navController.popBackStack()) {
+                        // NOP
+                    }
+                    routeStore.clearBackStack()
                 }
+                navOptions
             }
         }
 
         try {
-            navController.navigate(transition, args)
+            navController.navigate(destinationId, args, navOptions)
         } catch (e: IllegalArgumentException) {
             log.error("This appears to be a bug in navController", e)
             navController.navigate(destinationId, args)
         }
     }
 
-    private fun findTransitionId(@IdRes from: Int, @IdRes to: Int): Int? {
-        return when (Pair(from, to)) {
-            Pair(R.id.fragment_locked, R.id.fragment_filter) -> R.id.action_locked_to_filter
-            Pair(R.id.fragment_null, R.id.fragment_filter) -> R.id.action_to_filter
-            Pair(R.id.fragment_null, R.id.fragment_locked) -> R.id.action_to_locked
+    private fun findTransitionId(@IdRes from: Int, @IdRes destination: Int): Int? {
+        return when (from to destination) {
+            R.id.fragment_locked to R.id.fragment_filter_backdrop -> R.id.action_to_filter
+            R.id.fragment_filter_backdrop to R.id.fragment_locked -> R.id.action_to_locked
```

Comment:
Nit: use `src` and `dest` here. `from` and `destination` is mixing your metaphors, like comparing `chalk` and `oranges`.